src/entity/src/user.js

```diff
@@ -273,10 +273,8 @@ export default class User {
     return request.execute()
       .then(response => response.data)
       .then((data) => {
-        if (credentials[socialIdentityAttribute]) {
-          data[socialIdentityAttribute] = credentials[socialIdentityAttribute];
-        }
-
+        data = assign(data, credentials);
+        delete data.password;
```

Comment:
`credentials` could contain the password if the user uses `kinveyAuth` to login with. `delete data.password` removes the password property from the data object.